yammbs/_molecule.py

```diff
@@ -39,3 +40,12 @@ def _molecule_with_conformer_from_smiles(
     molecule.add_conformer(Quantity(conformer, "angstrom"))
 
     return molecule
+
+
+@lru_cache
+def _smiles_to_inchi_key(smiles: str) -> str:
+    from openff.toolkit import Molecule
+
+    return Molecule.from_mapped_smiles(smiles, allow_undefined_stereo=True).to_inchi(
```

Comment:
Just a note that this is the inchi not the inchikey as the function name suggests.

Reply:
Oops - I'm pretty sure it's what we'd want, though? i.e. the names of the functions/models/etc should be updated, but the method call `Molecule.to_inchi` is correct?

Reply:
I think Inchi or inchi key is fine, I like to use inchi key in qcsubmit as its fixed length but they should give the same result when checking if two molecules are the same.